connect/runtime/src/main/java/org/apache/kafka/connect/storage/KafkaStatusBackingStore.java

```diff
@@ -82,25 +86,46 @@
 
     private static final String TASK_STATUS_PREFIX = "status-task-";
     private static final String CONNECTOR_STATUS_PREFIX = "status-connector-";
+    private static final String TOPIC_STATUS_PREFIX = "status-topic-";
+    private static final String TOPIC_STATUS_SEPARATOR = ":connector-";
 
     public static final String STATE_KEY_NAME = "state";
     public static final String TRACE_KEY_NAME = "trace";
     public static final String WORKER_ID_KEY_NAME = "worker_id";
     public static final String GENERATION_KEY_NAME = "generation";
 
+    public static final String TOPIC_STATE_KEY = "topic";
+    public static final String TOPIC_NAME_KEY = "name";
+    public static final String TOPIC_CONNECTOR_KEY = "connector";
+    public static final String TOPIC_TASK_KEY = "task";
+    public static final String TOPIC_DISCOVER_TIMESTAMP_KEY = "discoverTimestamp";
+
     private static final Schema STATUS_SCHEMA_V0 = SchemaBuilder.struct()
             .field(STATE_KEY_NAME, Schema.STRING_SCHEMA)
             .field(TRACE_KEY_NAME, SchemaBuilder.string().optional().build())
             .field(WORKER_ID_KEY_NAME, Schema.STRING_SCHEMA)
             .field(GENERATION_KEY_NAME, Schema.INT32_SCHEMA)
             .build();
 
+    private static final Schema TOPIC_STATUS_VALUE_SCHEMA_V0 = SchemaBuilder.struct()
+            .field(TOPIC_NAME_KEY, Schema.STRING_SCHEMA)
+            .field(TOPIC_CONNECTOR_KEY, Schema.STRING_SCHEMA)
+            .field(TOPIC_TASK_KEY, Schema.INT32_SCHEMA)
+            .field(TOPIC_DISCOVER_TIMESTAMP_KEY, Schema.INT64_SCHEMA)
+            .build();
+
+    private static final Schema TOPIC_STATUS_SCHEMA_V0 = SchemaBuilder.map(
+            Schema.STRING_SCHEMA,
+            TOPIC_STATUS_VALUE_SCHEMA_V0
+    ).build();
+
     private final Time time;
     private final Converter converter;
     private final Table<String, Integer, CacheEntry<TaskStatus>> tasks;
     private final Map<String, CacheEntry<ConnectorStatus>> connectors;
+    private final ConcurrentMap<String, ConcurrentMap<String, TopicStatus>> topics;
 
-    private String topic;
+    private String statusTopic;
```

Comment:
note: That's a simple rename, because the member variable `topic` is easy to be masked by a local variable with this new feature.